auto_submit/lib/service/validation_service.dart

```diff
@@ -304,26 +307,57 @@ Exception: ${exception.message}
       // errors.
       final graphql.GraphQLClient client = await config.createGitHubGraphQLClient(slug);
 
-      final graphql.QueryResult result = await client.mutate(
-        graphql.MutationOptions(
-          document: mergePullRequestMutation,
-          variables: <String, dynamic>{
-            'id': id,
-            'oid': sha,
-            'title': '${queryResult.repository!.pullRequest!.title} (#$number)',
-          },
-        ),
-      );
-      if (result.hasException) {
-        final String message = 'Failed to merge pr#: $number with ${result.exception}';
+      bool mergeSuccessful = false;
+      int i = 1;
+      while (i <= Config.backoffAttempts) {
```

Comment:
If possible use the retry module rather than implementing your own. This helps to have consistent retry logic across the application. e.g. https://cs.opensource.google/flutter/cocoon/+/main:app_dart/lib/src/service/datastore.dart;l=42?q=retry&ss=flutter%2Fcocoon

Reply:
Ah cool! Okay will do. Thanks for the info.

Reply:
Done.